File: TIA/tia_final_project/src/com/crdzbird/models/City.java

package com.crdzbird.models;

/**
 *
 * @author crdzbird
 */

import java.util.Objects;

public class City implements Comparable<City> {
    
    private final String name;
    private final float x;
    private final float y;

    public City(String name, float x, float y) {
        this.name = name;
        this.x = x;
        this.y = y;
    }

    public String getName() {
        return name;
    }

    public float getX() {
        return x;
    }

    public float getY() {
        return y;
    }

    public static double euclideanDistance(City c1, City c2) {
        double dx = c2.getX() - c1.getX();
        double dy = c2.getY() - c1.getY();
        return Math.sqrt(dx * dx + dy * dy);
    }

    public static double manhattanDistance(City c1, City c2) {
        return Math.abs(c1.getX() - c2.getX()) + Math.abs(c1.getY() - c2.getY());
    }

    @Override
    public int compareTo(City o) {
        return name.compareTo(o.getName());
    }

    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }
        if (!(obj instanceof City)) {
            return false;
        }
        City other = (City) obj;
        return Objects.equals(name, other.name);
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }

    @Override
    public String toString() {
        return "City {name=".concat(name).concat(", x=").concat(String.valueOf(x)).concat(", y=").concat(String.valueOf(y)).concat("}\n");
    }
}

File: TIA/tia_genetic_memetic/lib/models/city.dart
import 'dart:math' as math;

class City implements Comparable<City> {
  final String name;
  final double x;
  final double y;

  City(this.name, this.x, this.y);

  String get getName => name;

  double get getX => x;

  double get getY => y;

  static double euclideanDistance(City c1, City c2) {
    double dx = c2.x - c1.x;
    double dy = c2.y - c1.y;
    return math.sqrt(dx * dx + dy * dy);
  }

  static double manhattanDistance(City c1, City c2) {
    return (c1.x - c2.x).abs() + (c1.y - c2.y).abs();
  }

  @override
  int compareTo(City other) {
    return name.compareTo(other.name);
  }

  @override
  bool operator ==(Object other) {
    if (identical(this, other)) return true;

    if (other is City) {
      return name == other.name;
    }

    return false;
  }

  @override
  int get hashCode => name.hashCode;

  @override
  String toString() {
    return 'City {name: $name, x: $x, y: $y}';
  }
}
